inc/item_operatingsystem.class.php

```diff
@@ -0,0 +1,379 @@
+      echo $header_begin.$header_top.$header_end;
+
+      if ($number) {
+         foreach ($os as $data) {
+            $link         = NOT_AVAILABLE;
+            $assocID      = $data["assocID"];
+
+            $linkname = $data['name'];
+            if ($_SESSION["glpiis_ids_visible"] || empty($data["name"])) {
+               $linkname = sprintf(__('%1$s (%2$s)'), $linkname, $data["assocID"]);
+            }
+            $link = Toolbox::getItemTypeFormURL(self::getType());
+            $name = "<a href=\"".$link."?id=".$data["assocID"]."\">".$linkname."</a>";
+
+            echo "<tr class='tab_bg_1'>";
+            if ($canedit
+                && ($withtemplate < 2)) {
+               echo "<td width='10'>";
+               Html::showMassiveActionCheckBox(__CLASS__, $data["assocID"]);
+               echo "</td>";
+            }
+            echo "<td class='center'>{$name}</td>";
+            echo "<td class='center'>{$data['version']}</td>";
+            echo "<td class='center'>{$data['architecture']}</td>";
+            echo "<td class='center'>{$data['servicepack']}</td>";
+
+            echo "</tr>";
+            $i++;
+         }
+         echo $header_begin.$header_bottom.$header_end;
+      }
+
+      echo "</table>";
+      if ($canedit && $number && ($withtemplate < 2)) {
+         $massiveactionparams['ontop'] = false;
+         Html::showMassiveActions($massiveactionparams);
+         Html::closeForm();
+      }
+      echo "</div>";
+
+   }
+
+   function getConnexityItem($itemtype, $items_id, $getFromDB=true, $getEmpty=true,
+                             $getFromDBOrEmpty=true) {
+      //overrided to set $getFromDBOrEmpty to true
+      return parent::getConnexityItem($itemtype, $items_id, $getFromDB, $getEmpty, $getFromDBOrEmpty);
+   }
+
+   function showPrimaryForm($options = []) {
+      //overrided to set expected values for new item
+      $fields = $options;
+      if (isset($fields['id']) && $fields['id'] == 0) {
+         unset($fields['id']);
+      }
+      foreach ($fields as $field => $value) {
+         $this->fields[$field] = $value;
+      }
+      parent::showPrimaryForm($options);
+   }
+
+
+   function showForm($ID, $options=array()) {
+      global $DB;
+
+      $colspan = 4;
+
+      echo "<div class='center'>";
+
+      $this->initForm($ID, $this->fields);
+      $this->showFormHeader(['formtitle' => false]);
+
+      $rand = mt_rand();
+
+      echo "<tr class='headerRow'><th colspan='".$colspan."'>";
+      echo __('Operating system');
+      echo Html::hidden('itemtype', ['value' => $this->fields['itemtype']]);
+      echo Html::hidden('items_id', ['value' => $this->fields['items_id']]);
+      echo "</th></tr>";
+
+      echo "<tr class='tab_bg_1'>";
+      echo "<td><label for='dropdown_operatingsystems_id$rand'>".__('Name')."</label></td>";
+      echo "<td>";
+      OperatingSystem::dropdown(array('value' => $this->fields["operatingsystems_id"], 'rand' => $rand));
```

Comment:
Here and the following dropdowns, could we use the short notation (```[...]```) for arrays like the other further in this function ?